tsconfig.eslint.json

```diff
@@ -5,10 +5,11 @@
     "allowJs": true
   },
   "include": [
-    "./eslint.config.js",
-    "packages/**/*",
-    "examples/**/*",
-    "e2e/**/*",
-    "./playwright.config.ts"
+    "eslint.config.js",
+    "packages/",
+    "examples/",
+    "examples/44_cloudflare/.wrangler/**/*",
+    "e2e/",
+    "playwright.config.ts"
```

Comment:
Can you explain how it works and reference we prefer these syntax?

Reply:
@dai-shi 

https://www.typescriptlang.org/tsconfig/#include

> These filenames are resolved relative to the directory containing the tsconfig.json file.

By this, omitting `./` will act based on the current directory and will not affect the file path.

> If the last path segment in a pattern does not contain a file extension or wildcard character, then it is treated as a directory, and files with supported extensions inside that directory are included (e.g. .ts, .tsx, and .d.ts by default, with .js and .jsx if [allowJs](https://www.typescriptlang.org/tsconfig/#allowJs) is set to true).

According to this, `packages/**/*` can be simplified to `packages/` because the last path segment is `packages`.
Similarly, `examples/**/*` can be written simply as `examples/` and `e2e/**/*` as `e2e/`.

Both `e2e` and `e2e/` are considered directories because they do not contain extensions or wildcards. `e2e` and `e2e/` are technically identical, but I thought adding `/` could more clearly indicate that they are directory paths.

Because `.wrangler` is a file extension that TypeScript does not recognize, `.wrangler` files inside the directory are automatically ignored, even if we include a directory like `exemples/`. Therefore I added `examples/44_cloudflare/.wrangler/**/*` to let Typescript recognize it separately.

Reply:
Interesting. The Cloudflare dev server generates a `examples/44_cloudflare/.wrangler/types/runtime.d.ts` file and I guess that happens to pass Waku's eslint rules when it is included in the typescript project.

I think it still might be better to just ignore eslint from the contents of any `.wrangler` directory. It is outside of our control. It doesn't exist until the 44_cloudflare example dev server is run so it doesn't cause an eslint failure on a fresh checkout of the code, but it is convenient when developing to not have to remove before running `pnpm test`.